src/components/app/details/triggerView/cdMaterial.tsx

```diff
@@ -1,231 +1,852 @@
+        }
+
+        return this.state.selectedMaterial
+            ? this.state.selectedMaterial.wfrId
+            : this.props.material?.find((_mat) => _mat.isSelected)?.wfrId
+    }
+
+    getDeploymentConfigDetails() {
+        this.setState({
+            checkingDiff: true,
+        })
+        const { appId, pipelineId } = this.props
+        Promise.allSettled([
+            getRecentDeploymentConfig(appId, pipelineId),
+            getLatestDeploymentConfig(appId, pipelineId),
+            getSpecificDeploymentConfig(appId, pipelineId, this.getWfrId()),
+        ]).then(
+            ([recentDeploymentConfigRes, latestDeploymentConfigRes, specificDeploymentConfigRes]: {
+                status: string
+                value?: any
+                reason?: any
+            }[]) => {
+                const _recentDeploymentConfig = processResolvedPromise(recentDeploymentConfigRes)
+                const _specificDeploymentConfig = processResolvedPromise(specificDeploymentConfigRes)
+                const _diffOptions = checkForDiff(_recentDeploymentConfig, _specificDeploymentConfig)
+
+                this.setState({
+                    recentDeploymentConfig: _recentDeploymentConfig,
+                    latestDeploymentConfig: processResolvedPromise(latestDeploymentConfigRes),
+                    specificDeploymentConfig: _specificDeploymentConfig,
+                    diffFound: Object.values(_diffOptions).some((d) => d),
+                    diffOptions: _diffOptions,
+                    checkingDiff: false,
+                })
+            },
+        )
+    }
+
+    async getSecurityModuleStatus(): Promise<void> {
+        try {
+            const { result } = await getModuleInfo(ModuleNameMap.SECURITY)
+            if (result?.status === ModuleStatus.INSTALLED) {
+                this.setState({ isSecurityModuleInstalled: true })
+            }
+        } catch (error) {}
+    }
+
+    renderGitMaterialInfo(matInfo) {
+        return (
+            <>
+                {matInfo.map((mat) => {
+                    let _gitCommit: GitTriggers = {
+                        Commit: mat.revision,
+                        Author: mat.author,
+                        Date: mat.modifiedTime,
+                        Message: mat.message,
+                        WebhookData: JSON.parse(mat.webhookData),
+                        Changes: [],
+                        GitRepoUrl: '',
+                        GitRepoName: '',
+                        CiConfigureSourceType: '',
+                        CiConfigureSourceValue: '',
+                    }
+
+                    return (
+                        <div className="bcn-0 pt-12 br-4 pb-12 en-2 bw-1 m-12">
+                            <GitCommitInfoGeneric
+                                materialUrl={mat.url}
+                                showMaterialInfo={false}
+                                commitInfo={_gitCommit}
+                                materialSourceType={''}
```

Comment:
Instead of sending empty strings can't we make these props optional so that we wont pass unnecessary props

Reply:
As discussed, I'm not making any changes around this as it may increase the scope of development & testing unnecessarily. We'll cover this as part of enhancement efforts.